### ISSUE TITLE
[RTA] Research Task - Backfill tests for retrospective feed generation functions

### ISSUE THREAD

**Issue by mrtopsyt** [cal-itp/data-analyses#1600]
## Research Task

Currently, there aren't tests for retrospective transit accessibility feed generation. Since this codebase is expected to grow and have products that are dependent on it, tests should be backfilled to validate that basic functionality works. Skipping functions that impute unrealistic times for now, since they might be moved upstream or changed in a way that makes it easier to write tests alongside the changes.

- Create test data as follows as necessary to implement the below tests
  - GTFS schedule feeds 
  - RT stop times tables
- Create tests as follows:
  - `make_retrospective_feed_single_date` returns a GTFS object with:
      - All of the existing tables within the GTFS specification from the schedule input
      - All tables except `stop_times` and `trips` are identical to the input GTFS object.
  - On a trip that exists with identical stop-times in both the schedule and RT table, `make_retrospective_feed_single_date` returns a new schedule feed that includes that trip exactly as it is in both tables
   - On a trip that does not exist in the schedule feed but does in the RT table, `make_retrospective_feed_single_date` raises a `KeyError`
  - On a trip that does exist in the schedule feed but does not in the RT table, `make_retrospective_feed_single_date` returns a new schedule feed that
    - does not include that trip id anywhere in `trips.txt`
    - does not include a trip with those exact stop times
  - On a trip that has complete and monotonically-increasing stop times in both the schedule feed and RT table, but where the stop times differ `make_retrospective_feed_single_date` returns a new schedule feed that includes that trip id with the stop times from the RT table at the same `stop_sequence`
  - For a trip that exists in both the input and output of `make_retrospective_feed_single_date`, the resulting stop times have identical data except `arrival_time` and `departure_time` at the same `stop_sequence`.
  - Confirm that outputs are copied correctly - altering a value in an arbitrary row of both `stop_times`, `trips` and an arbitrary table in the output of `make_retrospective_feed_single_date` does not alter the value in the original schedule input
  - Check that, when `make_retrospective_feed_single_date` is called with `validate=True`, assertion errors are raised as follows:
    - When, at the same `trip_id` and `stop_sequence`, `stop_id` is different between the rt table and the schedule feed
    - When, at the same `trip_id` and `stop_sequence`, the `arrival_time` matches the schedule arrival time from the RT table
* **Relevant PR** (if applicable):  

* **Question or Goal**: Validate RTA feed generation code, and allow for further development

* **Data Required**:
  * Currently have: Test infrastructure in #1599 
  * Need:  

* **Research Required**:  na

* **Metrics**: Test success rate?

* **Expected Outputs / Findings**: Tests!
